File: bigdime-libraries/bigdime-hdfs-lib/src/main/java/io/bigdime/libs/hdfs/WebHdfsGetFileStatusResponse.java
package io.bigdime.libs.hdfs;

import java.util.HashMap;
import java.util.Map;

import org.codehaus.jackson.annotate.JsonAnyGetter;
import org.codehaus.jackson.annotate.JsonAnySetter;
import org.codehaus.jackson.annotate.JsonIgnore;
import org.codehaus.jackson.annotate.JsonProperty;
import org.codehaus.jackson.annotate.JsonPropertyOrder;

@JsonPropertyOrder({ "FileStatus" })
public class WebHdfsGetFileStatusResponse {

	@JsonProperty("FileStatus")
	private FileStatus fileStatus;
	@JsonIgnore
	private Map<String, Object> additionalProperties = new HashMap<String, Object>();

	/**
	 * 
	 * @return The fileStatus
	 */
	@JsonProperty("FileStatus")
	public FileStatus getFileStatus() {
		return fileStatus;
	}

	/**
	 * 
	 * @param fileStatus
	 *            The FileStatus
	 */
	@JsonProperty("FileStatus")
	public void setFileStatus(FileStatus fileStatus) {
		this.fileStatus = fileStatus;
	}

	@JsonAnyGetter
	public Map<String, Object> getAdditionalProperties() {
		return this.additionalProperties;
	}

	@JsonAnySetter
	public void setAdditionalProperty(String name, Object value) {
		this.additionalProperties.put(name, value);
	}

}

File: bigdime-libraries/bigdime-hdfs-lib/src/main/scala/io/bigdime/libs/hdfs/WebHdfsGetFileStatusResponse.scala
package io.bigdime.libs.hdfs

import com.fasterxml.jackson.annotation.{JsonIgnore, JsonIgnoreProperties, JsonProperty, JsonPropertyOrder}

/**
  * Created by neejain on 3/17/17.
  */
@JsonPropertyOrder(Array("FileStatus"))
@JsonIgnoreProperties(ignoreUnknown = true)
case class WebHdfsGetFileStatusResponse(@JsonProperty("FileStatus") val FileStatus: FileStatus, @JsonIgnore val additionalProperties: Map[String, Any])

@JsonPropertyOrder(Array("accessTime", "blockSize", "childrenNum", "fileId", "group", "length", "modificationTime", "owner", "pathSuffix", "permission", "replication", "type"))
@JsonIgnoreProperties(ignoreUnknown = true)
case class FileStatus(@JsonProperty("accessTime") val accessTime: Long, @JsonProperty("blockSize") val blockSize: Long, @JsonProperty("childrenNum") val childrenNum: Long, @JsonProperty("fileId") val fileId: Long, @JsonProperty("group") val group: String, @JsonProperty("length") val length: Long, @JsonProperty("modificationTime") val modificationTime: Long, @JsonProperty("owner") val owner: String, @JsonProperty("pathSuffix") val pathSuffix: String, @JsonProperty("permission") val permission: String, @JsonProperty("replication") val replication: Long, @JsonProperty("type") val `type`: String, @JsonIgnore val additionalProperties: Map[String, Any])

